### PR TITLE
Feature/te 294 tcl rcp

### PR DESCRIPTION
**Pull request by gunther-bachmann**
please review

### CODE REVIEW

File: aml/org.testeditor.aml.model/plugin.xml [hunk 1]
```diff
@@ -10,7 +10,15 @@
       <!-- @generated model -->
       <package
             uri="org.testeditor.aml.model"
-            class="org.testeditor.aml.model.ModelPackage"
+            class="org.testeditor.aml.ModelPackage"
```
Comment (franzbecker) [inline aml/org.testeditor.aml.model/plugin.xml]: This should be removed.

File: rcp/org.testeditor.tcl.rcp.platform.feature/pom.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (franzbecker) [inline rcp/org.testeditor.tcl.rcp.platform.feature/pom.xml]: This should be a common, shared project (nothing tcl specific).

File: rcp/org.testeditor.tcl.rcp.product/.project [hunk 3]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (franzbecker) [inline rcp/org.testeditor.tcl.rcp.product/.project]: Same here, should be the testeditor rcp product, not the tcl.rcp.product

File: rcp/org.testeditor.tcl.rcp.product/src/org/testeditor/tcl/rcp/product/Application.xtend [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+	 * @see IApplication#stop()
+	 */
+	override stop() {
+		if (!PlatformUI.isWorkbenchRunning)
+			return
+		val workbench = PlatformUI.workbench
+		val display = workbench.display
+		display.syncExec[
+			if (!display.isDisposed) 
+				workbench.close
```
Comment (franzbecker) [inline rcp/org.testeditor.tcl.rcp.product/src/org/testeditor/tcl/rcp/product/Application.xtend]: Please add brackets here `{`

File: rcp/org.testeditor.tcl.rcp.product/src/org/testeditor/tcl/rcp/product/Application.xtend [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+			display.dispose
+		}
+	}
+
+	/* (non-Javadoc)
+	 * @see IApplication#stop()
+	 */
+	override stop() {
+		if (!PlatformUI.isWorkbenchRunning)
+			return
```
Comment (franzbecker) [inline rcp/org.testeditor.tcl.rcp.product/src/org/testeditor/tcl/rcp/product/Application.xtend]: Please add brackets here `{`

File: rcp/org.testeditor.tcl.rcp.product/src/org/testeditor/tcl/rcp/product/ApplicationWorkbenchAdvisor.xtend [hunk 6]
```diff
@@ -0,0 +1,75 @@
+package org.testeditor.tcl.rcp.product
```
Comment (franzbecker) [inline rcp/org.testeditor.tcl.rcp.product/src/org/testeditor/tcl/rcp/product/ApplicationWorkbenchAdvisor.xtend]: Please format this class

File: rcp/org.testeditor.tcl.rcp.product/src/org/testeditor/tcl/rcp/product/ApplicationWorkbenchWindowAdvisor.xtend [hunk 7]
```diff
@@ -0,0 +1,51 @@
+package org.testeditor.tcl.rcp.product
```
Comment (franzbecker) [inline rcp/org.testeditor.tcl.rcp.product/src/org/testeditor/tcl/rcp/product/ApplicationWorkbenchWindowAdvisor.xtend]: Please format this class

File: rcp/org.testeditor.tcl.rcp.product/src/org/testeditor/tcl/rcp/product/ApplicationWorkbenchWindowAdvisor.xtend [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+	override createActionBarAdvisor(IActionBarConfigurer configurer) {
+		new ApplicationActionBarAdvisor(configurer)
+	}
+
+	override preWindowOpen() {
+		windowConfigurer => [
+			initialSize=new Point(800, 600)
+			showCoolBar=false
+			showStatusLine=false
+			title="TCL" // $NON-NLS-1$
```
Comment (franzbecker) [inline rcp/org.testeditor.tcl.rcp.product/src/org/testeditor/tcl/rcp/product/ApplicationWorkbenchWindowAdvisor.xtend]: Title should be `TestEditor`

File: rcp/org.testeditor.tcl.rcp.product/src/org/testeditor/tcl/rcp/product/NewWizard.xtend [hunk 9]
```diff
@@ -0,0 +1,38 @@
+package org.testeditor.tcl.rcp.product
```
Comment (franzbecker) [inline rcp/org.testeditor.tcl.rcp.product/src/org/testeditor/tcl/rcp/product/NewWizard.xtend]: Please format this class

File: releng/org.testeditor.aml.releng.target.updatesite/site.xml [hunk 10]
```diff
@@ -1,16 +1,15 @@
...
    <feature url="features/org.eclipse.emf.sdk_2.11.1.v20150806-0404.jar" id="org.eclipse.emf.sdk" version="2.11.1.v20150806-0404"/>
    <feature url="features/org.eclipse.emf.ecore.xcore.sdk_1.3.1.v20150806-0546.jar" id="org.eclipse.emf.ecore.xcore.sdk" version="1.3.1.v20150806-0546"/>
    <feature url="features/org.eclipse.emf.mwe2.language.sdk_2.8.1.v201508030411.jar" id="org.eclipse.emf.mwe2.language.sdk" version="2.8.1.v201508030411"/>
    <feature url="features/org.eclipse.emf.mwe2.runtime.sdk_2.8.1.v201508030321.jar" id="org.eclipse.emf.mwe2.runtime.sdk" version="2.8.1.v201508030321"/>
    <feature url="features/org.eclipse.xtend.sdk_2.9.0.v201512010527.jar" id="org.eclipse.xtend.sdk" version="2.9.0.v201512010527"/>
    <feature url="features/org.eclipse.xtext.sdk_2.9.0.v201512010527.jar" id="org.eclipse.xtext.sdk" version="2.9.0.v201512010527"/>
-   <feature url="features/org.eclipse.equinox.p2.rcp.feature_1.2.101.v20150826-1318.jar" id="org.eclipse.equinox.p2.rcp.feature" version="1.2.101.v20150826-1318">
+   <feature url="features/org.eclipse.equinox.p2.rcp.feautre_1.2.101.v20150826-1318.jar" id="org.eclipse.equinox.p2.rcp.feature" version="1.2.101.v20150826-1318">
```
Comment (franzbecker) [inline releng/org.testeditor.aml.releng.target.updatesite/site.xml]: Is the JAR really named this way or is this a typo? `feautre_1`

File: releng/org.testeditor.aml.releng.target.updatesite/site.xml [hunk 11]
```diff
@@ -1,16 +1,15 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <site>
-   <feature url="features/org.testeditor.aml.releng.target.thirdparty_1.0.0.qualifier.jar" id="org.testeditor.aml.releng.target.thirdparty" version="1.0.0.qualifier"/>
+   <feature url="features/org.testeditor.aml.releng.target.thirdparty_1.0.0-SNAPSHOT.jar" id="org.testeditor.aml.releng.target.thirdparty" version="1.0.0"/>
```
Comment (franzbecker) [inline releng/org.testeditor.aml.releng.target.updatesite/site.xml]: Why is this renamed to `SNAPSHOT`?

File: releng/org.testeditor.aml.releng.target.updatesite/site.xml [hunk 12]
```diff
@@ -30,5 +29,20 @@
...
+   <!-- in order to create rcp projects with ide support using this target platform, the following rcp specific jars must be added -->
+   <feature url="features/org.eclipse.rcp.source_4.5.1.v20150904-0015.jar" id="org.eclipse.rcp.source" version="4.5.1.v20150904-0015">
+      <category name="eclipse-platform"/>
+   </feature>
+   <feature url="features/org.eclipse.rcp_4.5.1.v20150904-0015.jar" id="org.eclipse.rcp" version="4.5.1.v20150904-0015">
+      <category name="eclipse-platform"/>
+   </feature>
+   <!-- get {e|j}git -->
+   <feature url="features/org.eclipse.jgit_4.0.3.201509231615-r.jar" id="org.eclipse.jgit" version="4.0.3.201509231615-r"/>
+   <feature url="features/org.eclipse.egit_4.0.3.201509231615-r.jar" id="org.eclipse.egit" version="4.0.3.201509231615-r"/>
```
Comment (franzbecker) [inline releng/org.testeditor.aml.releng.target.updatesite/site.xml]: Looks like this line is redundant

File: tcl/org.testeditor.tcl.dsl/src/org/testeditor/tcl/dsl/TclStandaloneSetup.java [hunk 13]
```diff
@@ -30,8 +30,8 @@ public static void doSetup() {
 	
 	@Override
 	public Injector createInjectorAndDoEMFRegistration() {
-	    EPackage.Registry.INSTANCE.put(TslPackage.eINSTANCE.getNsURI(), TslPackage.eINSTANCE);
-	    EPackage.Registry.INSTANCE.put(TclPackage.eINSTANCE.getNsURI(), TclPackage.eINSTANCE);
+	    // EPackage.Registry.INSTANCE.put(TslPackage.eNS_URI, TslPackage.eINSTANCE);
```
Comment (franzbecker) [inline tcl/org.testeditor.tcl.dsl/src/org/testeditor/tcl/dsl/TclStandaloneSetup.java]: This line should be removed

**Comment by kpanier** [comment]
Launching the product from the product doesn't work . Some dependencies can't be resolved.

**Comment by gunther-bachmann** [comment]
Jenkins build now complete (and green)
https://ci.testeditor.org/view/AML/job/AML-feature-TE-294-build-complete/
